### PR TITLE
:art: Allow fiction coordinates for simulation and other functions.

### PR DESCRIPTION
**Pull request by Drewniok**
## Description

This PR makes it possible to use fiction coordinates for simulation.

## Checklist:

- [x] The pull request only contains commits that are related to it.
- [x] I have added appropriate tests and documentation.
- [ ] I have made sure that all CI jobs on GitHub pass.
- [ ] The pull request introduces no new warnings and follows the project's style guidelines.

### CODE REVIEW

File: include/fiction/utils/layout_utils.hpp [hunk 1]
```diff
@@ -408,42 +415,71 @@ CoordinateType random_coordinate(CoordinateType coordinate1, CoordinateType coor
  * @param cell_se The southeast SiQAD cell defining the ending point of the area.
  * @return A vector containing all cells within the specified area.
  */
-[[nodiscard]] inline std::vector<siqad::coord_t> all_sidbs_in_spanned_area(const siqad::coord_t& cell_nw,
-                                                                           const siqad::coord_t& cell_se) noexcept
+template <typename CoordinateType>
+[[nodiscard]] inline std::vector<CoordinateType> all_sidbs_in_spanned_area(const CoordinateType& cell_nw,
```
Comment (marcelwa) [inline include/fiction/utils/layout_utils.hpp:419]: This function doesn't require SiDBs anywhere. Can we maybe change its name to reflect that?

File: include/fiction/utils/layout_utils.hpp [hunk 2]
```diff
@@ -408,42 +415,71 @@ CoordinateType random_coordinate(CoordinateType coordinate1, CoordinateType coor
  * @param cell_se The southeast SiQAD cell defining the ending point of the area.
```
Comment (marcelwa) [inline include/fiction/utils/layout_utils.hpp:415]: There are still some SiQAD namings in the comments here. Maybe do a quick CTRL + SHIFT + F for "siqad" across the project to see what else there is to replace.

File: include/fiction/utils/layout_utils.hpp [hunk 3]
```diff
@@ -408,42 +415,71 @@ CoordinateType random_coordinate(CoordinateType coordinate1, CoordinateType coor
...
-
-    std::vector<siqad::coord_t> all_cells{};
-    all_cells.reserve(total_cell_count);
-
-    auto current_cell = cell_nw;
-
-    // collect all cells in the area (spanned by the nw `north-west` and se `south-east` cell) going from top to down
-    // from left to right.
-    while (current_cell <= cell_se)
+    // for siqad coordinates
```
Comment (marcelwa) [inline include/fiction/utils/layout_utils.hpp:422]: I noticed inconsistent usage of "siqad" vs. "SiQAD" in comments. In the code, we have naming conventions of course. In the comments, we should be able to call the tool by its original spelling, no?

**Review by marcelwa — CHANGES_REQUESTED** [review]
Many thanks for the thorough changes. I have only a few comments.

File: include/fiction/algorithms/simulation/sidb/quickexact.hpp [hunk 4]
```diff
@@ -125,12 +125,12 @@ class quickexact_impl
...
                     if constexpr (has_get_sidb_defect_v<Lyt>)
                     {
-                        charge_distribution_surface charge_layout{static_cast<sidb_defect_cell_clk_lyt_siqad>(layout)};
+                        charge_distribution_surface charge_layout{static_cast<Lyt>(layout)};
                         conduct_simulation(charge_layout, base_number);
                     }
                     else
                     {
-                        charge_distribution_surface charge_layout{static_cast<sidb_cell_clk_lyt_siqad>(layout)};
+                        charge_distribution_surface charge_layout{static_cast<Lyt>(layout)};
```
[the remark this answers is not in the document]
Comment (Drewniok) [reply include/fiction/algorithms/simulation/sidb/quickexact.hpp:133]: @marcelwa If I recall correctly, it is required for the Windows CIs to pass.

File: include/fiction/layouts/bounding_box.hpp [hunk 5]
```diff
@@ -67,7 +67,7 @@ class bounding_box_2d
             int32_t min_y = std::numeric_limits<int32_t>::max();
             int32_t max_y = std::numeric_limits<int32_t>::min();
 
-            uint8_t min_z = 0;
+            uint8_t min_z = 1;
```
[the remark this answers is not in the document]
Comment (Drewniok) [reply include/fiction/layouts/bounding_box.hpp:70]: @marcelwa bug fix

File: include/fiction/technology/sidb_nm_position.hpp [hunk 6]
```diff
@@ -28,12 +29,20 @@ template <typename Lyt>
...
+        return std::make_pair(x, y);
+    }
+    else
+    {
+        const auto cell_in_siqad = siqad::to_siqad_coord(c);
+        const auto x             = (cell_in_siqad.x * sp.lat_a) * 0.1;
+        const auto y             = (cell_in_siqad.y * sp.lat_b + cell_in_siqad.z * sp.lat_c) * .1;
+        return std::make_pair(x, y);
+    }
 }
```
[the remark this answers is not in the document]
Comment (Drewniok) [reply include/fiction/technology/sidb_nm_position.hpp:46]: Wow! That's wonderful!

[the remark this answers is not in the document]
Comment (Drewniok) [reply include/fiction/technology/sidb_nm_position.hpp:46]: @marcelwa Unfortunately, it does not work.

**Review by marcelwa — APPROVED** [review]

**Pull request merged**